src/codegen.cpp

```diff
@@ -237,7 +237,7 @@ void SonicParse_Statement_Assignment::generateCode ( std::ostream &o, Sonic_Code
             modify = true;
         }
 
-        for ( i=1; i < numWaveSymbols; i++ )
+        for ( int i=1; i < numWaveSymbols; i++ )

```

Comment:
I would rather see these changed just in one place like this:
````
int i;
for ( i=1; i < numWaveSymbols; i++ )
````
That would resolve all the following compiler errors with minimal code change.